index.js

```diff
@@ -34,8 +34,12 @@ const processor = (report, reporterOptions = {}, jestRootDir = null) => {
   // Ensure output path exists
   mkdirp.sync(path.dirname(outputPath));
 
+  // Clean ESC character, which is invalid XML and some libraries include in error messages
+  const xmlString = xml(jsonResults, { indent: '  ', declaration: true });
+  const cleanedXmlString = xmlString.replace(/\u001b/g, '');
```

Comment:
Instead of doing a simple string replace can you see if you can utilize something more generic like strip-ansi?

We use that package when generating the failure tag. So perhaps it needs to be included here or tweaked.

See https://github.com/jest-community/jest-junit/blob/master/utils/buildJsonResults.js#L77

Reply:
Thank you, it appears that `strip-ansi` strips only ANSI escape codes, which are ESC plus some characters. But it doesn't also strip plain ASCII control characters like ESC, which is illegal in the XML spec.

![Screenshot 2022-11-10 at 6 03 03 PM](https://user-images.githubusercontent.com/83245491/201246187-e09117a3-9145-48f1-af01-4486fa62d24a.png)

I failed to find a suitable package on the npm registry. I did find a snippet https://gist.github.com/john-doherty/b9195065884cdbfd2017a4756e6409cc from @john-doherty that appears to be more robust than my naive approach. I initially wanted to keep the scope of the change small, but if you prefer, I can inline a copy with attribution or work with the author to publish it as a separate package with appropriate license. @palmerj3, please just let me know as I'm happy to follow your preference.

Reply:
Cool thanks for clarifying. Your implementation looks fine to me so let's roll with that.

Running the CI job now. If all looks good I'll merge and cut a release in the morning.

Reply:
It looks like the checks passed.

Sorry for the ping. It's not urgent for us since we've already patched locally, but it looks like there's one other PR that intersects and a couple of downstream projects that could use this fix. Thanks for your attention here and hope you're doing better!